### PR TITLE
ocamlPackages.caqti: 1.7.0 -> 1.8.0

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

Update caqti to latest release, fixes regression from #165614 cc @vbgl 

Added `doCheck = true;` which needed `alcotest@1.5.0` so I upgraded that as well. Running `nixpkgs-review` currently, it should be posted shortly.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/alcotest/default.nix [hunk 1]
```diff
@@ -1,18 +1,20 @@
 { lib, buildDunePackage, fetchurl
-, astring, cmdliner, fmt, uuidm, re, stdlib-shims, uutf
+, astring, cmdliner, fmt, uuidm, re, stdlib-shims, uutf, ocaml-syntax-shims
```
Comment (bcc32) [inline pkgs/development/ocaml-modules/alcotest/default.nix:2]: This argument is unused.

Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/alcotest/default.nix:2]: It's used on line 16

Comment (bcc32) [reply pkgs/development/ocaml-modules/alcotest/default.nix:2]: Oh, sorry, you're right.  I was confused by GitHub's diff highlighting.  (More saturated background for words added on a line, but not for a completely-new line.  Weird.)

Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/alcotest/default.nix:2]: No problem, it wasn't too unlikely that I would have missed this 😂

**Comment by vbgl** [comment]
Thanks.

**Pull request merged**